### ISSUE TITLE
[indent] regression with typescript-eslint-parser@20.1.0

### ISSUE THREAD

**Issue by scottohara** [bradzacher/eslint-plugin-typescript#271]
This issue was originally posted over at https://github.com/eslint/typescript-eslint-parser/issues/586; but it was suggested that it be raised here as well. Please refer to the original issue for additional commentary/discussion.

**Repro**

```JSON
{
  "parser": "typescript-eslint-parser",
  "plugins": ["typescript"],
  "rules": {
    "indent": "off",
    "typescript/indent": ["error", "tab", {"VariableDeclarator": {"const": 3}}]
  }
}

```
(in the code below, ^ represents a tab character)

```TS
interface Foo {
  a: number;
  b: number;
}

const foo: Foo = {
^ ^ ^ ^ a: 1,
^ ^ ^ ^ b: 2
^ ^ ^ },
^ ^ ^ bar = 1;
```

**Expected Result**
No errors

**Actual Result**
In `typescript-eslint-parser@20.0.0`, code passes with no errors.

With `typescript-eslint-parser@20.1.0` onwards, the errors below occur:

```bash
7:1  error  Expected indentation of 1 tab but found 4   indent
8:1  error  Expected indentation of 1 tab but found 4   indent
9:1  error  Expected indentation of 0 tabs but found 3  indent
```

**Additional Info**
The code can be modified like below to pass, but clearly this is not what the config intends:

```TS
const foo: Foo = {
  a: 1,
  b: 2
},
      bar = 1
```

Vanilla JS code without a type annotation passes as expected:

```js
const foo = {
        a: 1,
		b: 2
      },
      bar = 1;
```

It also worth noting that the same issue occurs when using:
* spaces instead of tabs
* the main `indent` rule from `eslint` instead of the `typescript/indent` rule from `eslint-plugin-typescript`

eslint-plugin-typescript | typescript-eslint-parser | indent | typescript/indent
-- | -- | -- | --
0.14.0 | 20.0.0 | ✅ | N/A
1.0.0-rc.0 | 20.0.0 | ✅ | ✅
0.14.0 | 20.1.0+ | ❌ | N/A
1.0.0-rc.0 | 20.1.0+ | ❌ | ❌

**Versions**

| package                    | version   |
| -------------------------- | --------- |
| `eslint`                   | `5.11.1`  |
| `typescript`				 | `3.1.1`   |
| `eslint-plugin-typescript` | `0.14.0+` |
| `typescript-eslint-parser` | `20.1.0+` |

**Comment by bradzacher** [comment]
It is known that v20+ of the parser breaks when using plugin version 0.14. They released a number of breaking changes which we had to handle ourselves. IIRC the only supported version for plugin v0.14 was parser@^17.

For the plugin 1.0.0 rc - we have moved the parser dependency into our plugin, so that we can control which version users are using.

Could you please try to switch your parser to use our exposed version, and let me know if it's still an issue?

---

Excerpt from: https://github.com/bradzacher/eslint-plugin-typescript/releases/tag/1.0.0-rc.0

1) Update your config to use our exported parser:
```diff
 {
-    "parser": "typescript-eslint-parser",
+    "parser": "eslint-plugin-typescript/parser",
     "plugins": ["typescript"]
 }
```

2) remove your direct dependency on the parser:
```
$ yarn remove typescript-eslint-parser
$ npm uninstall typescript-eslint-parser
```

**Comment by scottohara** [comment]
Thanks @bradzacher.

Unfortunately, the problem is the same when using `"parser": "eslint-plugin-typescript/parser"`.

`package.json`
```json
"dependencies": {
  "eslint": "5.11.1",
  "eslint-plugin-typescript": "1.0.0-rc.0",
  "typescript": "3.1.1"
}
```

`.eslintrc.json`
```json
{
  "parser": "eslint-plugin-typescript/parser",
  "plugins": ["typescript"],
  "rules": {
    "indent": "off",
    "typescript/indent": ["error", "tab", {"VariableDeclarator": {"const": 3}}]
  }
}
```

`index.ts`
```ts
interface Foo {
  a: number;
  b: number;
}

const foo: Foo = {
        a: 1,
        b: 2
      },
      bar = 1;
```

`output`
```bash
7:1  error  Expected indentation of 1 tab but found 4   typescript/indent
8:1  error  Expected indentation of 1 tab but found 4   typescript/indent
9:1  error  Expected indentation of 0 tabs but found 3  typescript/indent
```
